### PR TITLE
Footer new

### LINKED ISSUE #12: Footer
- Three lists of links
- Replace icons for Twitter ([check X brand toolkit](https://about.twitter.com/en/who-we-are/brand-toolkit)) and YouTube
- **To do:** check what happens to current links "Cookies" and "Do Not Sell or Share My Personal Information"

### PR DESCRIPTION
**Pull request by kesiah**
Fix #12

I started changing the fill attribute in the SVGs files, so now we are using `<path fill="var(--color-icon, #000)" />` the CSS variable to be able to change the color. 
This update for the rest of the icons will be done in a followup PR ;)

Test URLs:
- Before: https://main--vg-macktrucks-com-rd--netcentric.hlx.page/drafts/tdziezyk/v2/hero
- After: https://footer-new--vg-macktrucks-com-rd--netcentric.hlx.page/drafts/tdziezyk/v2/hero

### CODE REVIEW

**Review by Lakshmishri — CHANGES_REQUESTED** [review]
@cogniSyb - should the logo in footer be clickable ? In current site it is clickable and navigates to homepage.

File: package-lock.json
[the remark this answers is not in the document]
Comment (avinashNChavan) [reply package-lock.json:1]: Updated

File: blocks/v2-footer/v2-footer.js [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+
+function findList(ele) {
+  if (ele.classList.contains(blockNameTruckList)) {
+    return ele;
+  }
+  return findList(ele.parentElement);
+}
+
+function toggleExpand(targetH3) {
+  const clickedColumn = findList(targetH3);
```
[the remark this answers is not in the document]
Comment (avinashNChavan) [reply blocks/v2-footer/v2-footer.js:57]: Fixed

File: blocks/v2-footer/v2-footer.css [hunk 2]
```diff
@@ -0,0 +1,482 @@
...
+/* Prefooter */
+.v2-prefooter {
+  background-color: var(--c-primary-gold);
+}
+
+.v2-prefooter ul {
+  padding: 8px 16px;
+}
+
+.v2-prefooter li {
```
[the remark this answers is not in the document]
Comment (avinashNChavan) [reply blocks/v2-footer/v2-footer.css]: Updated

File: blocks/v2-footer/v2-footer.js [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+        const submitButton = block.querySelector('input[type="submit"]');
+        const emailInput = block.querySelector('input[name="emailAddress"]');
+        const label = emailInput.parentElement.querySelector('label');
+        const emailAndSubmitContainer = createElement('span', ['email-and-submit-container']);
+
+        // change the submit button to arrow button
+        // and display it sticked to the right side of email input
+        if (submitButton && emailInput) {
+          const parent = emailInput.parentElement;
+          submitButton.value = '';
```
[the remark this answers is not in the document]
Comment (avinashNChavan) [reply blocks/v2-footer/v2-footer.js:202]: Added

File: blocks/v2-footer/v2-footer.js [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+
+    newFooter.append(truckListContent);
+  }
+
+  // Menu: social media + logo + menu list + newsletter form
+  if (footerMenu) {
+    const newMenu = createElement('div', blockNameMenu);
+
+    // Logo
+    const logo = footerMenu.querySelector('picture');
```
[the remark this answers is not in the document]
Comment (avinashNChavan) [reply blocks/v2-footer/v2-footer.js]: Updated

File: blocks/v2-footer/v2-footer.js [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+
+  // Menu: social media + logo + menu list + newsletter form
+  if (footerMenu) {
+    const newMenu = createElement('div', blockNameMenu);
+
+    // Logo
+    // const logo = footerMenu.querySelector('picture');
+    const logo = createElement('div');
+    const logoLink = createElement('a');
+    logoLink.href = 'https://www.macktrucks.com/';
```
Comment (Lakshmishri) [inline blocks/v2-footer/v2-footer.js:119]: you can do this `const logoLink = createElement('a', '',  { href:  'https://www.macktrucks.com/'});`

**Review by Lakshmishri — CHANGES_REQUESTED** [review]
I would like to move the contents to footer.js and footer.css as this would be actual footer now and there is no version 2 .
While doing so remove the function loadFooter from scripts.js and use loadFooter from franklinlib

**Review by Lakshmishri — CHANGES_REQUESTED** [review]
In Build block use `footer` instead of` v2-footer`, as its not loading the footer currently

File: blocks/v2-footer/v2-footer.js (hunk 5 above)
Comment (avinashNChavan) [reply blocks/v2-footer/v2-footer.js]: Fixed

File: blocks/v2-footer/v2-footer.css [hunk 6]
```diff
@@ -0,0 +1,506 @@
...
+    padding: 0;
+  }
+
+  .v2-footer-truck-list__wrapper {
+    margin: auto;
+    display: flex;
+    align-items: baseline;
+  }
+
+  .v2-footer-truck-list__title {
```
[the remark this answers is not in the document]
Comment (avinashNChavan) [reply blocks/v2-footer/v2-footer.css]: Updated

File: blocks/v2-footer/v2-footer.css [hunk 7]
```diff
@@ -0,0 +1,506 @@
...
+.v2-footer-menu__column a:hover {
+  text-decoration: underline;
+}
+
+/* Newsletter */
+.v2-footer-newsletter {
+  grid-area: newsletter;
+}
+
+.v2-footer-newsletter__title {
```
[the remark this answers is not in the document]
Comment (avinashNChavan) [reply blocks/v2-footer/v2-footer.css]: Updated

**Review by Lakshmishri — APPROVED** [review]

**Pull request merged**